### DISCUSSION TITLE
Not memory-mapped registers

### DISCUSSION THREAD

**Discussion by xinwu-cyber** [SystemRDL/systemrdl-compiler#226] [Q&A] [answered]
Hello PeakRDL team,

The registers in my current project are not conventional memory-mapped registers; rather, they are accessed via the I2C interface. This means that the register addresses correspond to I2C addresses. As a result, the addresses are not spaced like 0x0, 0x4, 0x8, 0xC but may increment sequentially, such as 0x0, 0x1, 0x2, and 0x3.

Defining these registers in SystemRDL is okay, but how can I configure the PeakRDL compiler to ignore address overlaps? Alternatively, is there a better way to use PeakRDL for this scenario?

**Comment by amykyta3** [comment]
The SystemRDL language is strictly defined to use byte-addressing, so there is no valid mechanism to ignore such overlaps without violating the standard.

SystemRDL defaults to assume 32-bit registers, but ff your I2C registers are actually only 8-bits wide, then you can use the `regwidth=8` property to set them as such.

**Reply by xinwu-cyber** [reply, marked as answer]
Thank you very much, now I'm using the user defined properties (UDP) to specify these special address, 

- io_type : the type of the access interface, it could be i2c, uart, spi and etc...
- io_addr : the user defined address

it can also meet my needs.
But I have to update the exporters to handle it.
